target/ppc/translate.c

```diff
@@ -7471,9 +7494,28 @@ static int times_4(DisasContext *ctx, int x)
 
 /* TODO: More TRANS* helpers for extra insn_flags checks. */
 
-
 #include "decode-insn32.c.inc"
 #include "decode-insn64.c.inc"
+
+/*
+ * Incorporate CIA into the constant when R=1.
+ * Validate that when R=1, RA=0.
+ */
+static bool resolve_PLS_D(DisasContext *ctx, arg_D *d, arg_PLS_D *a)
+{
+    d->rt = a->rt;
+    d->ra = a->ra;
+    d->si = a->si;
+    if (a->r) {
```

Comment:
Why create an arg_D copy of arg_PLS_D? Is the idea to create others resolve_[format]_D and they always generate a arg_D?

Reply:
The idea is to enable code sharing with the non-prefixed version of the instruction. It's not always `arg_D`, just a coincidence that all prefixed versions that could share code so far had "something:D" form.